galaxyxml/tool/parameters/__init__.py

```diff
@@ -241,11 +243,18 @@ def acceptable_child(self, child):
 class InputParameter(XMLParam):
     def __init__(self, name, **kwargs):
         # TODO: look at
-        self.mako_identifier = name
+        if "argument" in kwargs and kwargs['argument']:
+            self.flag_identifier = kwargs['argument'].lstrip()
+            self.num_dashes = len(kwargs['argument'])-len(self.flag_identifier) 
+            self.mako_identifier = _parse_name(name, kwargs['argument'])
+        else:
+            self.flag_identifier = name
+            self.mako_identifier = name
+
         # We use kwargs instead of the usual locals(), so manually copy the
         # name to kwargs
         if name is not None:
-            kwargs["name"] = name
+            kwargs = dict([("name", name)] + list(kwargs.items()) )
```

Comment:
That looks like an ordereddict, a bit. Should we just use that here?

Reply:
We can, but

- seems that kwargs preserves the order since py3.6 https://www.python.org/dev/peps/pep-0468/
- also dicts are now (insertion) order preserving. 
- also this does not save us from this "hack"

Reply:
Ok! I did not know that was a new feature, lovely.